crates/ruff_workspace/src/resolver.rs

```diff
@@ -276,7 +278,7 @@ pub fn python_files_in_path(
     paths: &[PathBuf],
     pyproject_config: &PyprojectConfig,
     transformer: &dyn ConfigurationTransformer,
-) -> Result<(Vec<Result<DirEntry, ignore::Error>>, Resolver)> {
+) -> Result<(Vec<Result<ResolvedFile, ignore::Error>>, Resolver)> {
```

Comment:
I had to introduce a new `ResolvedFile` type to know whether the file was explicitly passed to the CLI or not. I could have used `DirEntry::depth` but I found that it leaks internal details about `python_files_in_path`